File: Unity/1-Trimestre/Scripts_Unity/Scripts/Aceleracion/Ejercicio1.cs
using UnityEngine;

public class Ejercicio1 : MonoBehaviour
{
    private int speed = 5;
    private float accelerationX = 0;
    private float accelerationY = 0;
    private float accelerationSpeed = 0.0002f;

    // Update is called once per frame
    void Update()
    {
        float horizontalAxis = Input.GetAxisRaw("Horizontal");
        float verticalAxis = Input.GetAxisRaw("Vertical");

        if (accelerationX < horizontalAxis)
        {
            accelerationX += accelerationSpeed;
        } else if (accelerationX > horizontalAxis)
        {
            accelerationX -= accelerationSpeed;
        }
        
        if (accelerationY < horizontalAxis)
        {
            accelerationY += accelerationSpeed;
        } else if (accelerationY > horizontalAxis)
        {
            accelerationY -= accelerationSpeed;
        }

        Vector3 direction = (accelerationX * horizontalAxis) * Vector3.right + (accelerationY * verticalAxis) * Vector3.up;
        Vector3 velocity = direction * Time.deltaTime;
        Vector3 displacement = velocity * speed;

        transform.position += displacement;

        if (transform.position.x < 0)
        {
            displacement *= 2;
        }
    }
}

File: Programacion/Tema3.Java/Bateria_Ejercicios_Opcionales/Ejercicio_1.java
public class Ejercicio_1 {
    public static void main(String[] args) {

        int n = 10;

        for (int i = 0; i < n; i++) {
            System.out.print("0 ");
        }

        System.out.println();

        for (int i = 0; i < n - 2; i++) {
            System.out.print("0 ");

            for (int j = 0; j < n - 2; j++) {
                System.out.print("* ");
            }

            System.out.println("0 ");
        }

        for (int i = 0; i < n; i++) {
            System.out.print("0 ");

        }
    }
}
